### DISCUSSION TITLE
sys/cdef.h not found

### DISCUSSION THREAD

**Discussion by freud96** [bol-edu/HLS-SOC-Discussions#48] [Q&A] [answered]
I'm building software emulation on vck machine for a newly created application, I import all source files, finish building software emulation(kernel). I get makefile contains error due to "sys/cdefs.h" not found when I am building configuration for system_hw_link. I've seen in xilinx's discussion thread that I can do **echo CPATH="/usr/include/x86_64-linux-gnu"** to solve it; in my case (export CPATH="/usr/lib/x86_64-linux-gnu") where the libc6 is located, however I am still getting the error.

![g++ error](https://user-images.githubusercontent.com/79575484/235403228-bc45f453-b0d9-47a9-92f9-2076b3b91dde.JPG)

**Comment by kevinjantw** [comment, marked as answer]
> I'm building software emulation on vck machine for a newly created application, I import all source files, finish building software emulation(kernel). I get makefile contains error due to "sys/cdefs.h" not found when I am building configuration for system_hw_link. I've seen in xilinx's discussion thread that I can do **echo CPATH="/usr/include/x86_64-linux-gnu"** to solve it; in my case (export CPATH="/usr/lib/x86_64-linux-gnu") where the libc6 is located, however I am still getting the error.
> 
> ![g++ error](https://user-images.githubusercontent.com/79575484/235403228-bc45f453-b0d9-47a9-92f9-2076b3b91dde.JPG)

Hi,
I found cdefs.h was located in /usr/include/x86_64-linux-gnu/sys/cdefs.h.
Maybe you can try to set export CPATH="/usr/include/x86_64-linux-gnu", but I can not sure CPATH is the lib prefix in your makefile.
If above change is still not working, please reproduce and post errors log step-by-step.

**Reply by freud96** [reply]
Thanks, it worked.
